website/src/components/pageStateSelectors/BaselineSelector.tsx

```diff
@@ -3,46 +3,144 @@ import type { DateRangeOption } from '@genspectrum/dashboard-components/util';
+            <label className='form-control'>
+                <div className='label'>
+                    <span className='label-text'>
+                        {locationFilterConfig.label ?? locationFilterConfig.placeholderText}
+                    </span>
+                </div>
+                <GsLocationFilter
+                    fields={locationFilterConfig.locationFields}
+                    onLocationChange={onLocationChange}
+                    value={locationFilterConfig.initialLocation}
+                    placeholderText={locationFilterConfig.placeholderText}
+                ></GsLocationFilter>
+            </label>
+
+            <label className='form-control'>
+                <div className='label'>
+                    <span className='label-text'>{dateRangeFilterConfig.label ?? dateRangeFilterConfig.dateColumn}</span>
+                </div>
+                <GsDateRangeSelector
+                    lapisDateField={dateRangeFilterConfig.dateColumn}
+                    onDateRangeChange={onDateRangeChange}
+                    earliestDate={dateRangeFilterConfig.earliestDate}
+                    initialValue={dateRangeFilterConfig.initialDateRange}
+                    dateRangeOptions={dateRangeFilterConfig.dateRangeOptions}
+                ></GsDateRangeSelector>
+            </label>
+
+            <>
+                {baselineFilterConfigs?.map((config: BaselineFilterConfig) => {
+                    switch (config.type) {
+                        case 'date': {
+                            return (
+                                <label className='form-control' key={config.dateColumn}>
+                                    <div className='label'>
+                                        <span className='label-text'>{config.label ?? config.dateColumn}</span>
+                                    </div>
+                                    <GsDateRangeSelector
+                                        lapisDateField={config.dateColumn}
+                                        onDateRangeChange={(newDateRange) => {
+                                            const newBaselineFilterConfigs = baselineFilterConfigs.map((conf) => {
+                                                if (conf.type !== 'date') {
+                                                    return conf;
+                                                }
+
+                                                return conf.dateColumn === config.dateColumn
+                                                    ? { ...conf, initialDateRange: newDateRange }
+                                                    : conf;
+                                            });
+                                            if (onBaselineFilterConfigChange) {
+                                                onBaselineFilterConfigChange(newBaselineFilterConfigs);
+                                            }
+                                        }}
```

Comment:
IMO this looks more complicated than it needs to be. We could probably simplify this by having in some top level component:

```
const [dateFilters, setDateFilters] = useState(initialPageState.baseline.dateFilters);
const [textFilters, setTextFilters] = useState(initialPageState.baseline.textFilters);

const newPageState = useMemo(() => {
   return {
    datasetFilter: {
      dateFilters,
      textFilters,
      ...
    },
    ...
  };
}, ...);
```

and passing the state down to this component:
```suggestion
                                        onDateRangeChange={(newDateRange) => {
                                            setDateFilters(previousDateFilters => ({
                                                ...previousDateFilters,
                                                [config.dateColumn]: newDateRange,
                                            }))
                                        }}
```

That would make the config actual config (because it doesn't change) and simplify handling state separately (reducing the overhead of dragging the config with us when updating the state).

Reply:
Essentially a lot of the simplification could also be achieved by turning `BaselineFilterConfig[]` into `{ [dateColumn]: BaselineFilterConfig }` (thus having a simple lookup instead of a loop, but leaving state and config in a single object).

Reply:
I now changed it, so that we have the config and the values separately.